File: internal/port/logger.go
package port

type LogLevel string

const (
	LogDebug LogLevel = "debug"
	LogInfo  LogLevel = "info"
	LogWarn  LogLevel = "warn"
	LogError LogLevel = "error"
)

var levelOrder = map[LogLevel]int{
	LogDebug: 10,
	LogInfo:  20,
	LogWarn:  30,
	LogError: 40,
}

func IsLevelEnabled(configured, candidate LogLevel) bool {
	return levelOrder[candidate] >= levelOrder[configured]
}

type Logger interface {
	Level() LogLevel
	Debug(message string, meta ...map[string]any)
	Info(message string, meta ...map[string]any)
	Warn(message string, meta ...map[string]any)
	Error(message string, meta ...map[string]any)
}

File: src/core/ports/outbound/logger.ts
export type LogLevel = 'debug' | 'info' | 'warn' | 'error';

const levelOrder: Record<LogLevel, number> = { debug: 10, info: 20, warn: 30, error: 40 };

export interface Logger {
  readonly level: LogLevel;
  debug(message: string, meta?: Record<string, unknown>): void;
  info(message: string, meta?: Record<string, unknown>): void;
  warn(message: string, meta?: Record<string, unknown>): void;
  error(message: string, meta?: Record<string, unknown>): void;
}

export function isLevelEnabled(configured: LogLevel, candidate: LogLevel): boolean {
  return levelOrder[candidate] >= levelOrder[configured];
}
